tools/cli/main.cpp

```diff
@@ -12,246 +12,859 @@
+//    (--symtab-version <version>)
+//    ( - | <input_file>...)
+
+static struct arg_rex *xtract = arg_rex1(NULL, NULL, EXTRACT_COMMAND_NAME, NULL, REG_ICASE, NULL);
+
+static struct arg_file *xtract_out_f = arg_file0(OUTPUT_SHORT_OPT,
+                                                 OUTPUT_LONG_OPT, FILE_DATATYPE, OUTPUT_FILE_GLOSSARY);
+
+static struct arg_file *xtract_err_f = arg_file0(ERR_SHORT_OPT, ERR_LONG_OPT,
+                                                 FILE_DATATYPE, ERR_GLOSSARY);
+
+static struct arg_str *xtract_output_type = arg_str0(OUT_TYPE_SHORT_OPT,
+                                                     OUT_TYPE_LONG_OPT,
+                                                     TYPE_DATATYPE,
+                                                     OUT_TYPE_GLOSSARY1);
+
+
+static struct arg_file *xtract_catalog_fs = arg_filen(CATALOG_SHORT_OPT,
+                                                      CATALOG_LONG_OPT,
+                                                      FILE_DATATYPE,
+                                                      0,
+                                                      MAX_NO_INPUTS,
+                                                      CATALOG_GLOSSARY1);
+
+static struct arg_str *xtract_symtable_name = arg_str1(SYM_TABLE_NAME_SHORT_OPT,
+                                                       SYM_TABLE_NAME_LONG_OPT,
+                                                       NAME_DATATYPE,
+                                                       SYM_TABLE_NAME_GLOSSARY);
+
+static struct arg_str *xtract_symtable_version = arg_str1(SYM_TABLE_VERSION_SHORT_OPT,
+                                                          SYM_TABLE_VERSION_LONG_OPT,
+                                                          VERSION_DATATYPE,
+                                                          SYM_TABLE_VERSION_GLOSSARY);
+
+static struct arg_file *xtract_input_fs = arg_filen(NULL, NULL, FILE_DATATYPE, 0, MAX_NO_INPUTS, INPUT_GLOSSARY);
+
+static struct arg_end *xtract_end = arg_end(20);
+
+static void *xtract_argtable[] =
+        {xtract, xtract_out_f, xtract_err_f, xtract_output_type, xtract_catalog_fs, xtract_symtable_name,
+         xtract_symtable_version, xtract_input_fs, xtract_end};
+
+static int xtract_nerrors;
+
+static void **all_arg_tables[] = {argtable, proc_argtable, comp_argtable, xtract_argtable, help_argtable,
+                                 version_argtable};
+static int all_arg_tables_size = sizeof(all_arg_tables) / sizeof(all_arg_tables[0]);
+
+//
+// Printing functions
+//
+
+void ion_print_process_examples() {
+    std::string s =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+            "Read input.10n and pretty-print it to stdout.\n"
```

Comment:
for all of these examples, i'd prefer to see a `:` at the end of the description instead of a `.`